### DISCUSSION TITLE
how to recognize ’A < B < C> >’ without searching ’>>‘ (antlr4.9.3)

### DISCUSSION THREAD

**Discussion by ReverseJay** [antlr/antlr4#3991] [Q&A] [answered]
i was writing a generic recognition with nest relation, like A < B < C > >
```
grammar Test;

generic: ID '<' genericElement  '>' ;
genericElement: ID | generic ;

ID: CHAR+ ;
CHAR:  'a'..'z' | 'A'..'Z';
```

when parsing A < B < C > >(without blank)
i got a error message: line 1:5 mismatched input '>>' expecting '>'

i have no idea how to fix it XD

**Comment by kaby76** [comment, marked as answer]
Check your build environment. The grammar works fine, although the grammar does not recognize whitespace, so spaces and newlines cannot appear at all in the input. Output the tokens recognized when debugging a parser.
```
$ cat Test.g4
grammar Test;

generic: ID '<' genericElement  '>' ;
genericElement: ID | generic ;

ID: CHAR+ ;
CHAR:  'a'..'z' | 'A'..'Z';

Kenne@DESKTOP-DL44R7B MINGW64 ~/z/Generated-CSharp
$ cat in.txt
A<B<C>>
Kenne@DESKTOP-DL44R7B MINGW64 ~/z/Generated-CSharp
$ ./bin/Debug/net6.0/Test.exe in.txt -tokens
[@0,0:0='A',<3>,1:0]
[@1,1:1='<',<1>,1:1]
[@2,2:2='B',<3>,1:2]
[@3,3:3='<',<1>,1:3]
[@4,4:4='C',<3>,1:4]
[@5,5:5='>',<2>,1:5]
[@6,6:6='>',<2>,1:6]
[@7,7:6='<EOF>',<-1>,1:7]

CSharp 0 in.txt success 0.0200715
Total Time: 0.087664
```

**Reply by ReverseJay** [reply]
it works fine now, i'm not sure what had happened before, i was building in windows, and using intellij with a bunch of java code, maybe something went wrong... 
thanks for replying and suggestion

**Reply by ReverseJay** [reply]
actually i find what's going on, in my parser file, there is a parser rule
```
//expr ('<<'|'>>') expr //cause problem
expr('<''<'|'>''>') expr //correct way to do
```
so now it works fine
